### PR TITLE
Added Variable and Identity data groups

### PR DESCRIPTION
**Pull request by indradeep**
Created two new data groups as per spec.  I'm still dissatisfied with the levels of awareness that the Transformer needs to have about the data groups.... wish there was a better way.

### CODE REVIEW

File: blurr/core/record.py [hunk 1]
```diff
@@ -1,9 +1,8 @@
 class Record:
     def __init__(self, event_dict):
-        self._event_dict = event_dict
+        for k, v in event_dict.items():
+            self.__dict__[k] = v
 
-    def __getattr__(self, item):
-        if isinstance(self._event_dict[item], dict):
```
Comment (prashastk) [inline blurr/core/record.py]: Removed dict in a dict support?

File: blurr/core/syntax/dtc_streaming_schema.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ DataGroups: list(any(include('data_group_variable'), include('data_group_identit
 
 data_group_variable:
   Name: identifier()
-  Type: enum('ProductML:DTC:DataGroup:Variable')
+  Type: enum('ProductML:DTC:DataGroup:VariableAggregate')
```
Comment (prashastk) [inline blurr/core/syntax/dtc_streaming_schema.yml]: Should we just call it Variable?

File: blurr/core/transformer.py [hunk 3]
```diff
@@ -51,4 +52,5 @@ def __init__(self, store: Store, schema: TransformerSchema, identity: str,
 
     def finalize(self):
         for item in self.nested_items.values():
-            self.store.save(Key(self.identity, item.name), item.snapshot)
+            if not isinstance(item, VariableDataGroup):
+                self.store.save(Key(self.identity, item.name), item.snapshot)
```
Comment (prashastk) [inline blurr/core/transformer.py]: Maybe add a save function to the DataGroups which takes in a store? VariableDataGroup will do nothing with it.

File: blurr/core/record.py (hunk 1 above)
Comment (indradeep) [reply blurr/core/record.py]: Yes.  It adds items to it's own object dictionary.

File: blurr/core/transformer.py (hunk 3 above)
Comment (indradeep) [reply blurr/core/transformer.py]: This has been addressed in a future PR.

File: blurr/core/syntax/dtc_streaming_schema.yml (hunk 2 above)
Comment (indradeep) [reply blurr/core/syntax/dtc_streaming_schema.yml]: I'll leave it as is now... le'ts address naming in a separate round.

File: blurr/core/record.py (hunk 1 above)
Comment (prashastk) [reply blurr/core/record.py]: How does dict in a dict work then? source.some_dict_in_streaming_data.value?

Comment (indradeep) [reply blurr/core/record.py]: Fixed.  All nested dictionaries will also become records.

**Pull request merged**